Assets/HoloToolkit/SpatialMapping/Scripts/SpatialMappingSource.cs

```diff
@@ -70,77 +103,189 @@ protected GameObject AddSurfaceObject(Mesh mesh, string objectName, Transform pa
 
             surfaceObject.Renderer = surfaceObject.Object.GetComponent<MeshRenderer>();
             surfaceObject.Renderer.sharedMaterial = RenderMaterial;
+            surfaceObject.Renderer.enabled = (drawVisualMeshesOverride ?? SpatialMappingManager.Instance.DrawVisualMeshes);
+            surfaceObject.Renderer.shadowCastingMode = ((castShadowsOverride ?? SpatialMappingManager.Instance.CastShadows) ? ShadowCastingMode.On : ShadowCastingMode.Off);
 
-            SurfaceObjects.Add(surfaceObject);
+            surfaceObject.Collider = surfaceObject.Object.GetComponent<MeshCollider>();
 
-            return surfaceObject.Object;
+            // Reset the surface mesh collider to fit the updated mesh. 
+            // Unity tribal knowledge indicates that to change the mesh assigned to a
+            // mesh collider, the mesh must first be set to null.  Presumably there
+            // is a side effect in the setter when setting the shared mesh to null.
+            surfaceObject.Collider.sharedMesh = null;
+            surfaceObject.Collider.sharedMesh = surfaceObject.Filter.sharedMesh;
+
+            return surfaceObject;
         }
 
         /// <summary>
-        /// Updates an existing surface object.
+        /// Add the surface to <see cref="SurfaceObjects"/>.
         /// </summary>
-        /// <param name="surfaceGameObject">Game object reference to the surfaceObject.</param>
-        /// <param name="meshID">User specified ID for the mesh.</param>
-        /// <returns>True if successful</returns>
-        protected void UpdateSurfaceObject(GameObject surfaceGameObject, int meshID)
+        /// <param name="toAdd">The surface to add.</param>
+        protected void AddSurfaceObject(SurfaceObject toAdd)
         {
-            // If it's in the list, update it
-            for (int i = 0; i < SurfaceObjects.Count; ++i)
+            surfaceObjectsWriteable.Add(toAdd);
+
+            var handlers = SurfaceAdded;
```

Comment:
This one I'd like to keep as "var".  Same rationale as the other "handlers" one.